types/math-toolbox/index.d.ts

```diff
@@ -0,0 +1,582 @@
+ */
+declare function gcd(a: number, b: number): number;
+
+/**
+ * Compute the dot product of any pair of 2D vectors.
+ *
+ * @param  x0 - First x start position.
+ * @param  y0 - First y start position.
+ * @param  x1 - First x end position.
+ * @param  y1 - First y end position.
+ * @param  x2 - Second x start position.
+ * @param  y2 - Second y start position.
+ * @param  x3 - Second x end position.
+ * @param  y3 - Second y end position.
+ * @return Dot product.
+ */
+declare function dotProduct(
+    x0: number,
+    y0: number,
+    x1: number,
+    y1: number,
+    x2: number,
+    y2: number,
+    x3: number,
+    y3: number,
+): number;
+
+/**
+ * Linearly interpolate between two values (mix).
+ *
+ * @param  x - Start of the range in which to interpolate.
+ * @param  y - End of the range in which to interpolate.
+ * @param  r - Value to use to interpolate between x and y.
+ * @return Lerped value
+ * @see {@link https://www.opengl.org/sdk/docs/man/html/mix.xhtml}
+ */
+declare function lerp(x: number, y: number, r: number): number;
+
+/**
+ * Convert degrees to radians.
+ *
+ * @param  degrees - Degrees.
+ * @return Angel in radians.
+ */
+declare function degToRad(degrees: number): number;
+
+/**
+ * Convert radians to degrees.
+ *
+ * @param  radians - Radians.
+ * @return Angel in degrees.
+ */
+declare function radToDeg(radians: number): number;
+
+/**
+ * Calculates the linear parameter t that produces the interpolant value within the range [a, b].
+ *
+ * @param  a - The start value.
+ * @param  b - The end value.
+ * @param  v - The value.
+ * @return The result of the reverse interpolation.
+ * @see {@link https://docs.unity3d.com/ScriptReference/Mathf.InverseLerp.html}
+ */
+declare function inverseLerp(a: number, b: number, v: number): number;
+
+/**
+ * Linearly interpolates between x and y by a with no limit to a.
+ *
+ * @param  x - The start value.
+ * @param  y - The end value.
+ * @param  a - The interpolation between the two floats.
+ * @return The float value as a result from the linear interpolation.
+ * @see {@link https://docs.unity3d.com/ScriptReference/Mathf.LerpUnclamped.html}
+ */
+declare function lerpUnclamped(x: number, y: number, a: number): number;
+
+/**
+ * Calculates the shortest difference between two given angles given in degrees.
+ *
+ * @param  a - Current.
+ * @param  b - Target.
+ * @return The distance.
+ * @see {@link https://docs.unity3d.com/ScriptReference/Mathf.DeltaAngle.html}
+ */
+declare function deltaAngleDeg(a: number, b: number): number;
+
+/**
+ * Same as Lerp but makes sure the values interpolate correctly when they wrap around 360 degrees.
+ *
+ * @param  a - The start value.
+ * @param  b - The end value.
+ * @param  t - Value to inerpolate.
+ * @return The interpolated value.
+ * @see {@link https://docs.unity3d.com/ScriptReference/Mathf.LerpAngle.html}
+ */
+declare function lerpAngleDeg(a: number, b: number, t: number): number;
+
+export {
+    almostIdentity,
```

Comment:
Why the big export block instead of explicitly exporting each?

Reply:
umm no reason just how it was generated from jsdoc comments. Should I fix this?

Reply:
It seems like it'd be clearer and easier to modify if they were plain exports, IMO. Additionally, the duplicated exports seem odd; they are not documented in https://terkelg.github.io/math-toolbox/.

Reply:
okay will fix the exports and I not get the last part

Reply:
You have:

```
    deltaAngleDeg,
    deltaAngleDeg as deltaAngle,
```

But the docs page doesn't list `deltaAngle`, only `deltaAngleDeg`.

Reply:
yes in docs they are not documented but in code there are two exports
(code)[https://github.com/terkelg/math-toolbox/blob/master/src/delta-angle-deg.js]
```js
export { deltaAngleDeg, deltaAngleDeg as deltaAngle }
```